python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -453,33 +425,61 @@ def _deblend(self, difference, positiveFootprints, negativeFootprints):
             Deblended positive and negative polarity footprints measured on
             ``difference``.
         """
-        def makeFootprints(sources):
-            footprints = afwDetection.FootprintSet(difference.getBBox())
-            footprints.setFootprints([src.getFootprint() for src in sources])
-            return footprints
-
-        def deblend(footprints):
-            """Deblend a positive or negative footprint set,
-            and return the deblended children.
-            """
-            sources = afwTable.SourceCatalog(self.schema)
-            footprints.makeSources(sources)
-            self.deblend.run(exposure=difference, sources=sources)
-            self.setPrimaryFlags.run(sources)
-            children = sources["detect_isDeblendedSource"] == 1
-            sources = sources[children].copy(deep=True)
-            # Clear parents, so that measurement plugins behave correctly.
-            sources['parent'] = 0
-            return sources.copy(deep=True)
-
-        positives = deblend(positiveFootprints)
-        negatives = deblend(negativeFootprints)
-
-        sources = afwTable.SourceCatalog(self.schema)
-        sources.reserve(len(positives) + len(negatives))
-        sources.extend(positives, deep=True)
-        sources.extend(negatives, deep=True)
-        return sources, makeFootprints(positives), makeFootprints(negatives)
+        # Merge the positive and negative footprints, but don't grow them.
```

Comment:
I'll have to check the code here to double check, but I'm pretty sure that the footprints created by `SourceDetectionTask` have already been grown. So growing them again will create unnecessary blending.

Reply:
Yes, SourceDetectionTask grows the footprints. I'm pretty sure you're correct here, and that we've always been over-growing during the merge.

Which reminds me: if you're deprecating `doMerge`, you should also deprecate `growFootprint`, since that was only used in the old merge code.